### PR TITLE
Update remote-apis-sdks to latest

### PR DESCRIPTION
**Pull request by peterebden**
This rolls in a bunch of breaking changes from upstream.

All the stuff on my fork that we want has been upstreamed too so we can get off that \o/

### CODE REVIEW

File: src/remote/remote.go [hunk 1]
```diff
@@ -170,13 +170,14 @@ func (c *Client) initExec() error {
 	if err != nil {
 		return err
 	}
+	client.DefaultRPCTimeouts["default"] = time.Duration(c.state.Config.Remote.Timeout)
```
Comment (peterebden) [inline src/remote/remote.go:173]: I don't know if we can do something more sensible here. Or just lose the config option completely maybe?

File: src/remote/action.go [hunk 2]
```diff
@@ -27,27 +27,27 @@ import (
 func (c *Client) uploadAction(target *core.BuildTarget, isTest, isRun bool) (*pb.Command, *pb.Digest, error) {
 	var command *pb.Command
 	var digest *pb.Digest
-	err := c.uploadBlobs(func(ch chan<- *chunker.Chunker) error {
+	err := c.uploadBlobs(func(ch chan<- *uploadinfo.Entry) error {
```
Comment (Tatskaari) [inline src/remote/action.go:30]: I don't understand this new nomenclature but whatever

File: src/remote/action.go [hunk 3]
```diff
@@ -273,14 +273,14 @@ func (c *Client) uploadInputDir(ch chan<- *chunker.Chunker, target *core.BuildTa
 	}
 	if !isTest && target.Stamp {
 		stamp := core.StampFile(target)
-		chomk := chunker.NewFromBlob(stamp, int(c.client.ChunkMaxSize))
+		entry := uploadinfo.EntryFromBlob(stamp)
 		if ch != nil {
-			ch <- chomk
+			ch <- entry
```
Comment (Tatskaari) [inline src/remote/action.go:278]: nit: emtry

File: src/remote/remote.go (hunk 1 above)
Comment (Tatskaari) [reply src/remote/remote.go:173]: What's this used for? General timeouts for all RPC calls? I can't imagine many people setting this. The only reason I can think to have it configurable is if we need to increase/decrease based on the SLAs you've specced your environment for.

**Review by Tatskaari — APPROVED** [review]
Looks good to me. Thanks for that.

File: src/remote/action.go (hunk 2 above)
Comment (peterebden) [reply src/remote/action.go:30]: I think an `uploadinfo.Entry` is a more abstract thing you're going to upload, a `Chunker` breaks one of those into pieces to push it.

In the new version the former is more relevant here (you can't ask the Chunker for its digest, for example)

File: src/remote/action.go (hunk 3 above)
Comment (peterebden) [reply src/remote/action.go:278]: I miss `chomk`

**Pull request merged**